src/toil/batchSystems/kubernetes.py

```diff
@@ -154,10 +161,12 @@ def issueBatchJob(self, jobNode):
             # For docs, start at the root of the job hierarchy:
             # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Job.md
 
-            # Make a definition for the container's resource requirements
+            # Make a definition for the container's resource requirements.
+            # Don't let people request too-small amounts of memory or disk.
+            # Kubernetes needs some lower limit to run the pod at all without OOMing.
             requirements_dict = {'cpu': jobNode.cores,
-                                 'memory': jobNode.memory,
-                                 'ephemeral-storage': jobNode.disk}
+                                 'memory': max(jobNode.memory, 1024 * 1024 * 512),
```

Comment:
This should probably log a warning to the user if they've specified resources too low for our minimum, otherwise I could see this as as undocumented and confusing reconfiguration of resources.

Reply:
Do we want the user to raise their Toil job's resource requirement just because Kubernetes can't start up a Toil pod with so little memory? Should the memory requirement for a job cover just the job, or the job and also the Toil overhead necessary to run it?

I was getting OOMs in the tests that asked for 100 MB of memory, so I upped it right to 512. But I don't think the test code in the jobs was itself using anywhere near 100 MB of memory, and the tests run fine wherever else they run (which actually might not be many places, now that I am looking at the batch system tests).

Reply:
@adamnovak I am thinking more of someone who is possibly attempting to optimize by setting the runtime requirements lower and seeing it magically/silently run jobs with an undocumented default.

I think it's sensible, I'm just wondering what the best way to present this to the user is.  A warning just being logged would be pretty immediately obvious, but maybe this could be a note in the documentation?  Do we need documentation on running on kubernetes?

Reply:
We probably do need some Kubernetes documentation. It's not necessarily immediately obvious that you want to use the AWS job store, for example (or whatever cloud/network jobs tore is appropriate for where your nodes are), or that we have special AWS secret glue.